sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollerTests.java

```diff
@@ -336,132 +197,111 @@ public void subscribeToOnlyFinalEventSuccessfullyCompleteInNSecondsTest() throws
      * after few in-progress response. But the sdk client will stop polling in between
      * and subscriber should never get final successful response.
      **/
+    @Ignore("https://github.com/Azure/azure-sdk-for-java/issues/5809")
     @Test
-    public void subscribeToAllPollEventStopPollingAfterNSecondsTest() throws Exception {
-
-        PollResponse<CreateCertificateResponse> successPollResponse = new PollResponse<>(PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED, new CreateCertificateResponse("Created : Cert A"));
-        PollResponse<CreateCertificateResponse> inProgressPollResponse = new PollResponse<>(PollResponse.OperationStatus.IN_PROGRESS, new CreateCertificateResponse("Starting : Cert A"));
-
-        long totalTimeoutInMillis = 1000 * 1;
-        Duration pollInterval = Duration.ofMillis(100);
-
-        Function<PollResponse<CreateCertificateResponse>, Mono<PollResponse<CreateCertificateResponse>>> pollOperation =
-            createPollOperation(inProgressPollResponse,
-                successPollResponse, totalTimeoutInMillis - pollInterval.toMillis());
-
-        Poller<CreateCertificateResponse> createCertPoller = new Poller<>(pollInterval, pollOperation);
-        Thread t = new Thread() {
-            @Override
-            public void run() {
-                try {
-                    Thread.sleep(totalTimeoutInMillis / 2);
-                    createCertPoller.setAutoPollingEnabled(false);
-                } catch (Exception e) {
-                }
-            }
-        };
-        t.start();
-        Thread.sleep(totalTimeoutInMillis);
-        Assert.assertTrue(createCertPoller.getStatus() == OperationStatus.IN_PROGRESS);
-        Assert.assertFalse(createCertPoller.isAutoPollingEnabled());
+    public void subscribeToAllPollEventStopPollingAfterNSecondsTest() {
+        // Assert
+        Duration pollInterval = Duration.ofSeconds(1);
+        Duration waitTime = Duration.ofSeconds(3);
+        PollResponse<Response> success = new PollResponse<>(OperationStatus.SUCCESSFULLY_COMPLETED, new Response("Created: Cert A"));
+        PollResponse<Response> inProgress = new PollResponse<>(OperationStatus.IN_PROGRESS, new Response("Starting : Cert A"));
+
+        when(pollOperation.apply(any())).thenReturn(Mono.just(inProgress), Mono.just(inProgress), Mono.just(success));
+
+        // Act
+        Poller<Response> poller = new Poller<>(pollInterval, pollOperation, null,
+            ignored -> new PollResponse<Response>(OperationStatus.USER_CANCELLED, null));
+
+        // Assert
+        StepVerifier.create(poller.getObserver())
+            .expectNext(inProgress)
+            .then(() -> poller.setAutoPollingEnabled(false))
+            .expectNoEvent(waitTime)
+            .then(() -> poller.cancelOperation())
+            .verifyError();
```

Comment:
This usecase ,user stop polling after N second. I think user is not calling  `cancelOperation` . Should we remove the call to `cancelOperation()` here ?

Reply:
Fixed.